### PR TITLE
tests,spread.yaml: introduce an upgrade test, support/split into two suites for this

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #1366

File: spread.yaml [hunk 1]
```diff
@@ -47,30 +52,41 @@ prepare: |
...
+            apt purge -y snapd || true
         restore-each: |
             $SPREAD_PATH/tests/lib/reset.sh --reuse-core
+
+    tests/upgrade/:
+        summary: Tests for snapd upgrade
+        restore:
+            apt purge -y snapd || true
+        restore-each: |
+            $SPREAD_PATH/tests/lib/reset.sh
```
Comment (fgimenez) [inline spread.yaml]: Maybe use `$TESTSLIB` here? Also, are `SPREAD_PATH` and `PROJECT_PATH` different?

Comment (pedronis) [inline spread.yaml]: no, they are the same actually, put PROJECT_PATH is just an internal detail of spread.yaml, SPREAD_PATH is fine in tests (it's not going to change), I can use TESTSLIB here though

File: tests/upgrade/basic/task.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    # setup not to reexec
+    export SNAP_REEXEC=0
+    grep SNAP_REEXEC /etc/environment || echo "SNAP_REEXEC=0">>/etc/environment
+    cat /etc/environment
+
+    echo Do upgrade
+    apt install -y ${SPREAD_PATH}/../snapd*.deb
+
+    snapdver=$(snap --version|grep "snapd ")
+    [ "$snapdver" != "$prevsnapdver" ] || exit 1
```
Comment (fgimenez) [inline tests/upgrade/basic/task.yaml]: Not sure if the `|| exit 1` are needed

**Comment by fgimenez** [comment]
Two nitpicks, :+1: thx!

**Comment by mvo5** [comment]
Nice!

**Comment by pedronis** [comment]
@mvo5 @fgimenez thanks, merging this

**Pull request merged**